user-onboarding/src/components/app.js

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState } from 'react'
+import * as yup from 'yup'
+import axios from 'axios'
+
+import Form from './form'
+import schema from './formSchema'
+import Users from './users'
+
+const initialFormValues = {
+   first_name: '',
+   last_name: '',
+   email: '',
+   role: '',
+   password: '',
+   confirmPassword: '',
+   terms: false
+}
+
+const initialFormErrors = {
+   first_name: '',
+   last_name: '',
+   email: '',
+   role: '',
+   password: '',
+   confirmPassword: '',
+   terms: false
+}
+
+const initialDisabled = true
+const initialUsers = []
+
+function App() {
+   const [users, setUsers] = useState(initialUsers)
+   const [formValues, setFormValues] = useState(initialFormValues)
+   const [errors, setErrors] = useState(initialFormErrors)
+   const [disabled, setDisabled] = useState(initialDisabled)
+
+   const getUsers = () => {
+      axios
+         .get(`https://reqres.in/api/users`)
+         .then(res => {
+            setUsers(res.data.data)
+         })
+         .catch(err => {
+            console.log(err)
+         })
+   }
+
+   useEffect(() => {
+      getUsers()
+   }, [])
+
+
+   useEffect(() => {
+      schema.isValid(formValues).then(valid => {
+         setDisabled(!valid)
+      })
+   }, [formValues])
+
+   const changeInput = (name, value) => {
+      yup
+         .reach(schema, name)
+         .validate(value)
+         .then(() => {
+            setErrors({ ...errors, [name]: '' })
+         })
+         .catch(err => {
+            setErrors({ ...errors, [name]: err.errors })
+         })
+
+      setFormValues({ ...formValues, [name]: value })
+   }
+
+   const postNewUser = (newUser) => {
+      axios
+         .post(`https://reqres.in/api/users`, newUser)
+         .then(res => {
+            setUsers([res.data, ...users])
+            setFormValues(initialFormValues)
```

Comment:
You might want to put the setFormValues function in a .finally after the catch. So that even if this request errors and ends up in the catch statement you still clear the form